drivers/SmartThings/deepsmart/src/discovery.lua

```diff
@@ -1,12 +1,13 @@
 local log = require('log')
 local Wisers = require "deepsmart.wisers"
 local ssdp = require('ssdp')
-local utils = require('utils')
+local net_utils = require('st.net_utils')
+
 local disco = {}
 -----------------------
 disco.ssdp_discovery_callback = function(uuid, ip, is_add)
-  if (not utils.is_valid_ipv4(ip) or uuid == nil) then
-    log.warn('ip or uuid is invalid')
+  if (not net_utils.validate_ipv4_string(ip) or uuid == nil) then
+    log.warn(string.format('ip (%s) or uuid (%s) is invalid', (ip or "nil"), (uuid or "nil")))

```

Comment:
nit: nil will be formatted by `%s` as the string "nil"
```suggestion
    log.warn(string.format('ip (%s) or uuid (%s) is invalid', ip, uuid ))
```